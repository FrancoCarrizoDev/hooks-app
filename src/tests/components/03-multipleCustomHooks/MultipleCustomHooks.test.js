import React from 'react';
import {shallow} from 'enzyme'
import { MultipleCustomHooks } from '../../../components/03-examples/MultipleCustomHooks';
import { useFetch } from '../../../hooks/useFetch';
import { useCounter } from '../../../hooks/useCounter';
jest.mock('../../../hooks/useFetch')

describe('Test in MultimpleCustomHooks components', () => {



    test('should show correctly', () => {

        useFetch.mockReturnValue({
            data: null,
            loading: true,
            error: null
        })
        const wrapper = shallow(<MultipleCustomHooks/>)

        expect(wrapper).toMatchSnapshot();
    })

    test('should show the info', () => {
        
        useFetch.mockReturnValue({
            data: [{
                author: 'Fran',
                quote: 'Hello word!'
            }],
            loading: false,
            error: null
        })

        const wrapper = shallow(<MultipleCustomHooks/>)

        expect(wrapper.find('.alert').exists()).toBe(false)
        expect(wrapper.find('.mb-0').text().trim()).toBe('Hello word!')
        expect(wrapper.find('footer').text().trim()).toBe('Fran')
        
    })
    
})
