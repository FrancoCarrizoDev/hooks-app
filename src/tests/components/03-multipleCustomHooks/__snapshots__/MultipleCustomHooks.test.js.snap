// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Test in MultimpleCustomHooks components should show correctly 1`] = `
<div>
  <h1>
    BrakingBad Quotes
  </h1>
  <hr />
  <div
    className="alert alert-info text-center"
  >
    Loading...
  </div>
  <button
    className="btn btn-primary"
    onClick={[Function]}
  >
    Siguiente frase
  </button>
</div>
`;
